### PR TITLE
Properly determine kernel affinity in controller

### PR DESCRIPTION
**Pull request by pacevedom**

### CODE REVIEW

File: controllers/resources.go [hunk 1]
```diff
@@ -19,6 +19,10 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 )
 
+var (
+	affineRegex = regexp.MustCompile("\\n\\s+specialresource\\.openshift\\.io\\/kernel\\-affine")
```
Comment (qbarrand) [inline controllers/resources.go]: How about using backticks here for a raw string? That should avoid some escaping.

File: controllers/resources.go [hunk 2]
```diff
@@ -161,9 +165,9 @@ func ReconcileChartStates(ctx context.Context, r *SpecialResourceReconciler) err
 		step := nostate
 		step.Templates = append(nostate.Templates, stateYAML)
 
-		// We are kernel-affine if the yamlSpec uses {{.Values.kernelFullVersion}}
+		// We are kernel-affine if the yamlSpec uses kernel-affine label.
 		// then we need to replicate the object and set a name + os + kernel version
-		kernelAffine := strings.Contains(string(stateYAML.Data), ".Values.kernelFullVersion")
```
Comment (qbarrand) [inline controllers/resources.go:166]: We're not looking for the same string anymore, are we?

**Comment by zvonkok** [comment]
If an asset uses the `kernelVersion` than it should be flagged as kernel-affine, regardless of the annotation. It is an either or between `kernelVersion` and annotation.

**Comment by qbarrand** [comment]
@zvonkok I appreciate the idea, but using `.Values.kernelFullVersion` does not sound very robust to me - someone could be using `{{ with .Values }}` in template, and that usage would not be captured by the current logic. Wouldn't relying on the label be easier?

**Comment by pacevedom** [comment]
My thinking was along the lines of @qbarrand, making this explicit in the resource through the use of the (already) dedicated label. Also trying to skip comment lines or even an unused value that would render a workload affine when it was not the intention.

**Comment by zvonkok** [comment]
Yep makes sense, we should make it explicit, agree a rogue comment with kernelVersion could make something affine that actually is not affine.

**Comment by qbarrand** [comment]
/lgtm

@pacevedom should we write this in some documentation before we merge it?
/hold

**Comment by pacevedom** [comment]
@qbarrand will be working on docs in a separate task. Thanks.

/ok-to-test

**Comment by pacevedom** [comment]
/unhold

Already working on separate task for docs.

**Pull request merged**